pybandits/transfer.py

```
@@ -434,16 +532,34 @@ def _expand_with_template_weights(
+
+    # Validate hidden layer architecture changes for overlapping actions
+    for action_id in set(current_cmab.actions.keys()) & set(template_cmab.actions.keys()):
+        src_model = current_cmab.actions[action_id]
+        tgt_model = template_cmab.actions[action_id]
+        if isinstance(src_model, BaseModelMO):
+            if len(src_model.models) != len(tgt_model.models):
+                raise ValueError(
+                    f"Cannot expand action '{action_id}': number of objectives mismatch "
+                    f"(current: {len(src_model.models)}, template: {len(tgt_model.models)})."
+                )
+            for obj_idx, (src_obj, tgt_obj) in enumerate(zip(src_model.models, tgt_model.models)):
```

Ruff (0.15.7)

[warning] `zip()` without an explicit `strict=` parameter

Add explicit value for parameter `strict=`

(B905)